core/commands/object.go

```diff
@@ -274,16 +274,18 @@ It reads from stdin, and the output is a base58 encoded multihash.
 'ipfs object put' is a plumbing command for storing DAG nodes.
 It reads from stdin, and the output is a base58 encoded multihash.
 
-Data should be in the format specified by <encoding>.
-<encoding> may be one of the following:
+Data should be in the format specified by the --inputenc flag.
+--inputenc may be one of the following:
 	* "protobuf"
-	* "json"
+	* "json" (default)
 `,
 	},
 
 	Arguments: []cmds.Argument{
-		cmds.FileArg("data", true, false, "Data to be stored as a DAG object"),
-		cmds.StringArg("encoding", true, false, "Encoding type of <data>, either \"protobuf\" or \"json\""),
+		cmds.FileArg("data", true, false, "Data to be stored as a DAG object").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption("inputenc", "Encoding type of input data, either \"protobuf\" or \"json\""),
```

Comment:
I thought it would be nicer to have two flags, so you can define input and output encoding independently.